src/pages/api/add-participant.ts

```diff
@@ -0,0 +1,31 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getPersonByDiscordId, createPerson } from '../../lib/api';
+import { postClient } from '../../lib/sanity';
+
+export default async function post(req: NextApiRequest, res: NextApiResponse) {
+  const { body } = req;
+
+  let user = await getPersonByDiscordId(body.discordUser);
+
+  if (!user) {
+    user = await createPerson({
+      username: { current: body.discordUser },
+    });
+  }
+
+  try {
+    postClient
+      .patch(body.id)
+      .setIfMissing({ participants: [] })
+      .insert('after', 'participants[-1]', [
+        {
```

Comment:
Do we need to send all the three fields? Isn't it enough with just sending `_ref`?

Reply:
I added a _key because the Sanity docs recommend it, but I guess you're right about the _type field.
![image](https://user-images.githubusercontent.com/33439343/108396997-40eb7b80-71f6-11eb-9346-14660b01acbb.png)